src/utils/trpc-provider.tsx

```diff
@@ -0,0 +1,55 @@
+'use client'
+
+import { QueryClientProvider } from '@tanstack/react-query'
+import { getFetch, httpBatchLink, loggerLink } from '@trpc/client'
+import { useEffect, useState } from 'react'
+import superjson from 'superjson'
+import queryClient from './query-client'
+import { trpc } from './trpc'
+import { signOut, useSession } from 'next-auth/react'
+
+export const TrpcProvider = ({ children }: { children: React.ReactNode }) => {
+  const url = process.env.NEXT_PUBLIC_VERCEL_URL
```

Comment:
`NEXT_PUBLIC_VERCEL_URL` is not used by us since we don't deploy on Vercel. We need to explicitly use a url that is defined. `NEXTAUTH_URL` can be used in this case since that is a full url that is _required_

Reply:
Use this instead

```
function getBaseUrl() {
  if (typeof window !== 'undefined')
    // browser should use relative path
    return ''
  if (process.env.VERCEL_URL)
    // reference for vercel.com deployments
    return `https://${process.env.VERCEL_URL}`
  if (process.env.NEXTAUTH_URL)
    // reference for non-vercel providers
    return process.env.NEXTAUTH_URL
  // assume localhost
  return `http://localhost:${process.env.PORT ?? 3000}`
}
```